packages/gatsby-image/src/index.js

```diff
@@ -145,20 +145,32 @@ class Image extends React.Component {
 
     if (props.critical) {
       isVisible = true
+      imgLoaded = false
+      IOSupported = false
     }
 
+    const hasNoScript =  !(this.props.critical && !this.props.fadeIn)
+
     this.state = {
       isVisible,
       imgLoaded,
       IOSupported,
+      fadeIn,
+      hasNoScript,
+      seenBefore,
     }
 
+    this.imageRef = React.createRef()
```

Comment:
need a reference to the HTMLImageElement to evaluate the completed state - to detect if DOM finished loading image before the JS started